src/main/java/com/breskul/bibernate/persistence/Persistence.java

```diff
@@ -1,17 +1,52 @@
 package com.breskul.bibernate.persistence;
 
-import org.postgresql.ds.PGSimpleDataSource;
+import com.breskul.bibernate.config.PropertiesConfiguration;
+import com.breskul.bibernate.exception.BibernateException;
+import java.util.Objects;
 
-// TODO: javadoc
+/**
+ * The Persistence class provides a convenient way to create a Bibernate SessionFactory based on configuration properties. It retrieves
+ * connection details from a properties file using {@link PropertiesConfiguration} and initializes a {@link BibernateDataSource}
+ * accordingly. The default behavior is to use a generic DataSource, but you can specify a custom JDBC driver class if needed.
+ * <p>
+ * Usage Example:
+ * <pre>
+ *   {@code
+ *  SessionFactory sessionFactory = Persistence.createSessionFactory();
+ * }
+ * </pre>
+ */
 public class Persistence {
 
-  // TODO: read properties for data source
+  private static final String CONNECTION_URL = "bibernate.connection.url";
+  private static final String CONNECTION_USERNAME = "bibernate.connection.username";
+  private static final String CONNECTION_PASSWORD = "bibernate.connection.password";
+  private static final String CONNECTION_DRIVER_CLASS = "bibernate.connection.driver_class";
+
+  /**
+   * Creates a Hibernate SessionFactory based on configuration properties.
+   *
+   * @return The created Hibernate SessionFactory.
+   * @throws BibernateException If there is an issue with the DataSource or loading the JDBC driver class.
+   * @see BibernateDataSource
+   * @see SessionFactory
+   * @see PropertiesConfiguration
+   */
   public static SessionFactory createSessionFactory() {
-    // TODO: initialize data source
-    PGSimpleDataSource dataSource = new PGSimpleDataSource();
-    dataSource.setUrl("jdbc:postgresql://localhost:5432/postgres");
-    dataSource.setUser("postgres");
-    dataSource.setPassword("admin");
+    String url = Objects.requireNonNull(PropertiesConfiguration.getProperty(CONNECTION_URL));
```

Comment:
What do you think about to pass  PropertiesConfiguration to DataSource constructor  and return created DataSource instance based on properties configured?
We can introduce some interface for this purposes and pass the interface to constructor,  this interface can be implemented by other config strategies (xml, java) later. So that we would not need to duplicate code for retrieving properties and passing them to DataSource.